endpoints/getting-started-grpc/client/main.go

```diff
@@ -61,8 +68,33 @@ func main() {
 	defer conn.Close()
 	c := pb.NewGreeterClient(conn)
 
+	if *keyfile != "" {
+		log.Printf("Authenticating using Google service account key in %s", *keyfile)
+		sakey, err := ioutil.ReadFile(*keyfile)
+		if err != nil {
+			log.Fatalf("Unable to read service account key file %s: %s", *keyfile, err)
+		}
+
+		tokenSource, err := googleoauth2.JWTAccessTokenSourceFromJSON(sakey, *audience)
+		if err != nil {
+			log.Fatalf("Error building JWT access token source: %v", err)
+		}
+		jwt, err := tokenSource.Token()
+		if err != nil {
+			log.Fatalf("Unable to generate JWT token: %v", err)
+		}
+		*token = jwt.AccessToken
```

Comment:
This is OK for this program, but should probably check token validation before using it later on L96.

Reply:
What sort of validation are you thinking of?

Reply:
Expiry, mostly. But it looks like these tokens will never expire.

Still, the call to tokenSource.Token is cheap, and you'd generally want to call it for every outgoing request.

You'd need to restructure this code a bit to accommodate this suggestion. But it's just a suggestion.

Reply:
Ah, makes sense. Across all implementations I've seen so far, JWT tokens are generated with a TTL of 1h. The Go implementation hard-codes 1h (https://github.com/golang/oauth2/blob/master/google/jwt.go#L56).

Reply:
Could the comment instead say:

```
// NOTE: the generated JWT token has a 1h TTL.
// Make sure to refresh the token before it expires by calling TokenSource.Token() for each outgoing request.
// Calls to TokenSource.Token() are cheap.
```

(all caps on NOTE, too)

Reply:
I updated the note as you suggested, although I wanted to make clear that `TokenSource.Token()` calls are not cheap in general. This particular implementation is reasonably cheap but not negligible, and other implementations (eg. for 2-legged OAuth or ID token) might involve additional RPCs.